core/src/apps/cardano/sign_tx.py

```diff
@@ -171,68 +147,39 @@ def _process_outputs(self):
     def _build_witnesses(self, tx_aux_hash: str):
         witnesses = []
         for input in self.inputs:
-            _, node = derive_address_and_node(self.keychain, input.address_n)
-            message = (
-                b"\x01" + cbor.encode(self.protocol_magic) + b"\x58\x20" + tx_aux_hash
-            )
-            signature = ed25519.sign_ext(
-                node.private_key(), node.private_key_ext(), message
-            )
-            extended_public_key = (
-                remove_ed25519_prefix(node.public_key()) + node.chain_code()
-            )
-            witnesses.append(
-                [
-                    (input.type or 0),
-                    cbor.Tagged(24, cbor.encode([extended_public_key, signature])),
-                ]
-            )
-
-        return witnesses
-
-    @staticmethod
-    def compute_fee(input_coins_sum: int, outgoing_coins: list, change_coins: list):
-        outgoing_coins_sum = sum(outgoing_coins)
-        change_coins_sum = sum(change_coins)
-
-        return input_coins_sum - outgoing_coins_sum - change_coins_sum
+            node = self.keychain.derive(input.address_n)
+            message = b"\x58\x20" + tx_aux_hash
+
+            signature = ed25519.sign_ext(node.private_key(), node.private_key_ext(), message)
+
+            public_key = remove_ed25519_prefix(node.public_key())
+
+            witnesses.append([public_key, signature])
+
+        return {0: witnesses}
 
     def serialise_tx(self):
 
         self._process_outputs()
 
         inputs_cbor = []
```

Comment:
Perhaps inputs_to_serialize --- they are not in cbor anymore until the whole tx body is cborized.